mesonbuild/optinterpreter.py

```diff
@@ -48,7 +48,7 @@
 reserved_prefixes = ['cross_']
 
 def is_invalid_name(name: str, *, log: bool = True) -> bool:
-    if name in forbidden_option_names:
+    if mesonlib.OptionKey(name) in forbidden_option_names:
         return True
     pref = name.split('_')[0] + '_'
     if pref in forbidden_prefixes:
```

Comment:
You need to use `from_string` here, not just `OptionKey`. But actually, this code is all pretty silly given the OptionKey helpers, it could be simplified to:
```python
key = mesonlib.OptionKey.from_string(name)
if not key.is_project():
    return True
elif key.name.startswith('cross_') and log:
    mlog.deprecation(...)
return False
```

Reply:
As we discussed on IRC, this whole function is actually useless as the deprecation is invalid anyway. Instead the caller could be replaced with:
```python
key = mesonlib.OptionKey.from_string(name).evolve(subproject=self.subproject)
if not key.is_project():
    raise OptionError(...)
```

Reply:
Indeed! And `cross_` does not need to be reserved any more, we went with `build.` prefix a long time ago.